doc/langdef.md

```diff
@@ -167,13 +318,272 @@ type `type`, which is an expression by itself which in turn also has type
+is converted to CEL data when read from a message field, and CEL data is
+converted in the other direction when initializing a field. In general, protocol
+buffer data can be converted to CEL without error, but range errors are possible
+in the other direction.
+
+| Protocol Buffer Type             | CEL Type                               |
+| -------------------------------- | -------------------------------------- |
+| int32, int64, sint32, sint64, sfixed32, sfixed64    | `int`               |
+| uint32, uint64, fixed32, fixed64 | `uint`                                 |
+| float, double                    | `double`                               |
+| bool, string, bytes              | same                                   |
+| enum                             | `int`                                  |
+| repeated                         | `list`                                 |
+| map<K, V>                        | `map`                                  |
+| oneof                 | options expanded individually, at most one is set |
+| message                          | same                                   |
+
+Signed integers, unsigned integers, and floating point numbers are converted to
+the singular CEL type of the same sort. The CEL type is capable of expressing
+the full range of protocol buffer values. When converting from CEL to protocol
+buffers, an out-of-range CEL value results in an error.
+
+Boolean, string, and bytes types have identical ranges and are converted without
+error.
+
+Protocol buffer enum values are converted to the corresponding `int` value.
+Protocol buffer enum fields can accept any signed 32-bit number, values outside
+that range will raise an error.
+
+Repeated fields are converted to CEL lists of converted values, preserving the
+order. In the other direction, the CEL list elements must be of the right type
+and value to be converted to the corresponding protocol buffer type. Similarly,
+protocol buffer maps are converted to CEL maps, and CEL map keys and values must
+have the right type and value to be converted in the other direction.
+
+Oneof fields are represented by the translation of each of their options as a
+separate field, but at most one of these fields will be "set", as detected by
+the `has()` macro.  See [Macros](#macros).
+
+Since protocol buffer messages are first-class CEL values, message-valued fields
+are used without conversion.
+
+Every protocol buffer field has a default value, and there is no semantic
+difference between a field set to this default value, and an unset field. For
+message fields, there default value is just the unset state, and an unset
+message field is distinct from one set to an empty (i.e. all-unset) message.
+
+The `has()` macro (see [Macros](#macros)) tells whether a message field is set
+(i.e. not unset, hence not set to the default value). If an unset field is
+nevertheless selected, it evaluates to its default value, or if it is a message
```

Comment:
".., or ..." Why or? The default value of an unset message field is its default value.

Reply:
Per the council meeting, we want unset dynamic type fields (e.g. g.p.Int64Value) to evaluate to null.  So the default value of a protocol buffer is null, but we make an exception when accessing (non-dyanmic) message fields.